ts/Extensions/PatternFill.ts

```diff
@@ -794,6 +801,116 @@ function wrapSeriesGetColor(
     }
 }
 
+/**
+ * Scales the pattern element to the given scales.
+ * @private
+ *
+ * @param {SVGDOMElement} element
+ * Pattern element to scale.
+ *
+ * @param {number} scaleX
+ * X scale to apply on the element.
+ *
+ * @param {number} scaleY
+ * Y scale to apply on the element.
+ */
+function setPatternScale(
+    element: SVGDOMElement,
+    scaleX: number,
+    scaleY: number
+): void {
+    // Get the current patternTransform value.
+    let transform = attr(element, 'patternTransform') || '';
+
+    // Regular expression to match various transforms.
+    const scaleRegex = /scale\([^)]+\)/;
+    const newScale = `scale(${scaleX}, ${scaleY})`;
+
+    // Check if scale is already present.
+    if (scaleRegex.test(transform)) {
+        // Update existing scale value.
+        transform = transform.replace(scaleRegex, newScale);
+    } else {
+        // Add new scale value if not present.
+        transform = `${transform} ${newScale}`.trim();
+    }
+
+    // Set the updated patternTransform value.
+    attr(element, 'patternTransform', transform);
+}
+
+/**
+ * Scale patterns inversly to the series it's used in.
+ * Maintains a visual (1,1) scale regardless of size.
+ * @private
+ */
+function seriesAfterRenderPatternScaleCorrection(
+    this: Series
+): void {
+    const series = this;
+
+    // If not a series used in a map chart, skip it.
+    if (!series.chart.mapView) {
+        return;
+    }
+
+    const chart = series.chart,
+        renderer = chart.renderer,
+        patterns = renderer.patternElements;
+
+    // Only scale if we have patterns to scale. #19980
+    if (renderer.defIds?.length && patterns) {
+        // Filter for points which have patterns that don't use images assigned
+        // and has a group scale available.
+        series.points.filter(function (p: Point): boolean {
+            return (
+                p.graphic?.element.hasAttribute('fill') ||
+                p.graphic?.element.hasAttribute('color') ||
+                p.graphic?.element.hasAttribute('stroke')
+            ) &&
+            !(p?.options.color as any)?.pattern?.image &&
+            (p as any).group?.scaleX &&
```

Comment:
Can we avoid the `any` cast here by casting to the `MapPoint` type`
`(p as MapPoint).group?.scaleX`